filesystems/watcher.go

```diff
@@ -561,6 +563,43 @@ func parsecurrentOffsetBytes(name string, currentOffsetBytes []byte) int64 {
 	return currentOffsetInt
 }
 
+func (w *Watcher) concurrentUploadLogFile(ctx context.Context, logFileList []string) error {
+	// How many workers do we need
+	uploadWorkers := concurrentUploaders
+	if len(logFileList) < uploadWorkers {
+		uploadWorkers = len(logFileList)
+	}
+
+	g, ctx := errgroup.WithContext(ctx)
+	// Limit concurrency
+	sem := semaphore.NewWeighted(int64(uploadWorkers))
+	for _, fileToUpload := range logFileList {
+		fileToUploadCopy := fileToUpload
+		err := sem.Acquire(ctx, 1)
+		if err != nil {
+			continue
+		}
+
+		g.Go(func() error {
+			defer sem.Release(1)
+			if !CheckFileForStdio(fileToUploadCopy) {
+				if err := w.uploadLogfile(ctx, fileToUploadCopy, true); err != nil {
```

Comment:
I think it's better to use multierror.Group. The reason being is that if one of the uploaders fails, you don't want to cancel all of them:
https://pkg.go.dev/github.com/hashicorp/go-multierror#Group

Reply:
Done.